### PR TITLE
restrict game start to the landlord unless no landlord is determined

### PR DESCRIPTION
**Pull request by cai-jack**
the request is landlord should have the exclusive privilege to start the game during play. (currently, the end game notification indicates the landlord will start the game, but anyone can). Some newly minted landlord feels HER hard earned victory was not sufficiently rewarded without the exclusive privilege to start a new game:)

### CODE REVIEW

**Comment by rbtying** [comment]
Oh, interesting. this is just restricting the "start game" button?

The "starting the game" from the message was intended to be "first to draw" and "first to play", haha

**Comment by cai-jack** [comment]
yes:)

**Comment by cai-jack** [comment]
the drawing does start with the landlord in the UI, otherwise it starts with the first player

File: core/src/game_state.rs [hunk 1]
```diff
@@ -1778,11 +1778,23 @@ impl InitializePhase {
         }
     }
 
-    pub fn start(&self) -> Result<DrawPhase, Error> {
+    pub fn start(&self, id: PlayerID) -> Result<DrawPhase, Error> {
         if self.propagated.players.len() < 4 {
             bail!("not enough players")
         }
 
+        match self.propagated.landlord {
```
Comment (rbtying) [inline core/src/game_state.rs:1786]: maybe

```
if self.propagated.landlord.map(|l| l != id).unwrap_or(false) {
   bail!("Only the landlord can start the game")
}
```

Comment (cai-jack) [reply core/src/game_state.rs]: cool!

**Comment by cai-jack** [comment]
understood. It is a desire by our player to have the game match the actual physical play a bit closer :)

**Comment by cai-jack** [comment]
Looks like this feature is actually in demand in our circle. I turned it into an option. Hope this is acceptable to you. Thanks.

**Pull request merged**